MAPL_ExceptionHandling/MAPL_ErrorHandling.F90

```diff
@@ -0,0 +1,174 @@
+      character(*), intent(in) :: filename
+      integer, intent(in) :: line
+      integer, intent(out), optional :: rc
+
+      logical :: condition, fail
+      character(:), allocatable :: message
+      character(8) :: status_string
+
+      condition = (status == 0)
+      fail = .not. condition
+
+      if (fail) then
+         write(status_string,'(i0)') status
+         message = 'status=' // status_string
+         call MAPL_throw_exception(filename, line, message=message)
+      end if
+      ! Regardless of error:
+      if (present(rc)) rc = status 
+      
+   end subroutine MAPL_Return
+
+   logical function MAPL_RTRN(A,iam,line,rc)
+      integer,           intent(IN ) :: A
+      character*(*),     intent(IN ) :: iam
+      integer,           intent(IN ) :: line
+      integer, optional, intent(OUT) :: RC
+
+        MAPL_RTRN = .true.
+        if(A/=0) print '(A40,I10)',Iam,line
+        if(present(RC)) RC=A
+   end function MAPL_RTRN
+
+   logical function MAPL_VRFY(A,iam,line,rc)
+      integer,           intent(IN ) :: A
+      character*(*),     intent(IN ) :: iam
+      integer,           intent(IN ) :: line
+      integer, optional, intent(OUT) :: RC
+        MAPL_VRFY = A/=0
+        if(MAPL_VRFY)then
+          if(present(RC)) then
+            print '(A40,I10)',Iam,line
+            RC=A
+          endif
+        endif
+   end function MAPL_VRFY
+
+   logical function MAPL_ASRT(A,iam,line,rc)
+      logical,           intent(IN ) :: A
+      character*(*),     intent(IN ) :: iam
+      integer,           intent(IN ) :: line
+      integer, optional, intent(OUT) :: RC
+        MAPL_ASRT = .not.A
+        if(MAPL_ASRT)then
+          if(present(RC))then
+            print '(A40,I10)',Iam,LINE
+            RC=1
+          endif
+        endif
+   end function MAPL_ASRT   
+
+   logical function MAPL_ASRTt(A,text,iam,line,rc)
+      logical,           intent(IN ) :: A
+      character*(*),     intent(IN ) :: iam,text
+      integer,           intent(IN ) :: line
+      integer, optional, intent(OUT) :: RC
+        MAPL_ASRTt =   MAPL_ASRT(A,iam,line,rc)
+        if(MAPL_ASRTt) print *, text
+   end function MAPL_ASRTT
+
+   logical function MAPL_RTRNt(A,text,iam,line,rc)
+      integer,           intent(IN ) :: A
+      character*(*),     intent(IN ) :: text,iam
+      integer,           intent(IN ) :: line
+      integer, optional, intent(OUT) :: RC
+
+        MAPL_RTRNt = .true.
+        if(A/=0)then
+           print '(A40,I10)',Iam,line
+           print *, text
+        end if
+        if(present(RC)) RC=A
+
+   end function MAPL_RTRNT
+
+   logical function MAPL_VRFYt(A,text,iam,line,rc)
+      integer,           intent(IN ) :: A
+      character*(*),     intent(IN ) :: iam,text
+      integer,           intent(IN ) :: line
+      integer, optional, intent(OUT) :: RC
+        MAPL_VRFYt =  MAPL_VRFY(A,iam,line,rc)
+        if(MAPL_VRFYt) print *, text
+   end function MAPL_VRFYT
+
+   subroutine MAPL_Abort_comm_world
```

Comment:
Was there a specific driver to make this distinction?  Just curious.

Reply:
@tclune 
What distinction, the MAP_Abort? I thought the point was this error handling library sits at the lowest level. MAPL_Abort lives  MAPL_Comms.F90 so I would not use MAPL_Abort. I guess I should have removed MAPL_Abort from MAPL_Comms and just moved it here rather than duplicate the code.

Reply:
I'll just move MAPL_Abort into the erorrhandling module, nothing else was using it but the include files.